### PR TITLE
INTLY-6430: keycloak client updates from KEYCLOAK-13018

### PR DESCRIPTION
**Pull request by JameelB**
# Description
### **DO NOT MERGE**
This PR is a work in progress. This currently replaces the keycloak-client dependency to point to this [branch](https://github.com/JameelB/keycloak-client/tree/INTLY-6430) so that the following PR can be tested: https://github.com/integr8ly/keycloak-client/pull/6. This will need to be updated to point back to the correct dependency and update it to the correct vendor files.

- [x] Update the integr8ly/keycloak-client dependency with changes from https://github.com/integr8ly/keycloak-client/pull/6

- [x] Update rhsso user reconciler and tests to use new return format of `CreateIdentityProvider`, `CreateAuthenticatorConfig`, `CreateGroupClientRole` and `CreateGroupRealmRole`.

## Type of change
- [x] Dependency update

## Checklist
- [x] I have added/updated tests that prove my fix is effective or that my feature works
- [x] Verified independently on a cluster by reviewer

## Verification Steps
1. Run the operator using this branch
2. Ensure installation was successful

### CODE REVIEW

**Comment by pb82** [comment]
:eyes:

File: go.mod [hunk 1]
```diff
@@ -69,3 +69,5 @@ replace (
 replace github.com/openshift/api => github.com/openshift/api v3.9.1-0.20190904142825-922d36944de3+incompatible
 
 replace github.com/docker/docker => github.com/moby/moby v0.7.3-0.20190826074503-38ab9da00309 // Required by Helm
+
+replace github.com/integr8ly/keycloak-client => github.com/JameelB/keycloak-client v0.0.0-20200324114609-86ab7200ec70
```
Comment (pb82) [inline go.mod]: this needs to be removed after a new version has been released

Comment (JameelB) [reply go.mod]: yep, i'll need to update this PR after the changes in the Keycloak client get merged.

**Pull request merged**